tests/lib/util.py

```diff
@@ -48,9 +48,27 @@ def create_commit(
     **kwargs,
 ) -> str:
     options = ""
+
     if dt is not None:
+        # Store committer date in case it was set somewhere else
+        original_committer_date = os.environ.get("GIT_COMMITTER_DATE", None)
+
         options += f"--date {dt.isoformat()}"
-    return execute(cwd, f'git commit -m "{message}" {options}', **kwargs)
+        # The committer date is what is used to determine sort order for tags, etc
+        os.environ["GIT_COMMITTER_DATE"] = f"--date {dt.isoformat()}"
+
+    return_value = execute(cwd, f'git commit -m "{message}" {options}', **kwargs)
+
+    # Return committer date env var to prior value if set
+    if dt is not None:
+        if original_committer_date is None:
```

Comment:
If `execute()` will raise an exception, environment variable value will not be reverted. Use try.finally